src/GaussMods.jl

```diff
@@ -5,74 +5,34 @@ export GaussMod, AbstractMod
 # mod support for Gaussian integers until officially adopted into Base
 mod(z::Complex{<:Integer}, n::Integer) = Complex(mod(real(z), n), mod(imag(z), n))
 
-"""
-`GaussMod{m}(v)` creates a modular number in mod `m` with value `mod(v,m)`.
-`GaussMod{m}()` is equivalent to `GaussMod{m}(0)`.
-"""
-struct GaussMod{N} <: AbstractMod
-    val::Complex{Int}
-    function GaussMod(x::Complex{T},N::Int) where T<:Integer
-        @assert N>1 "modulus must be at least 2"
-        new{N::Int}(mod(x,N))
-    end
-end
-
-GaussMod{N}(x::Complex{T}=0+0im) where {N,T<:Integer} = GaussMod(x,N)
-GaussMod{N}(x::T) where {N,T<:Integer}  = GaussMod{N}(Complex{Int}(x))
-GaussMod{N}(x::S,y::T) where {N,S<:Integer,T<:Integer} = GaussMod{N}(x + y*im)
-GaussMod(x::Mod{N}) where N = GaussMod{N}(x.val)
-GaussMod{N}(x::Mod{N}) where N = GaussMod{N}(x.val)
-
-GaussMod{N}(x::Rational{T}) where {N,T<:Integer} = GaussMod{N}(numerator(x)) / GaussMod{N}(denominator(x))
-GaussMod{N}(x::Complex{Rational{T}}) where {N,T<:Integer} = GaussMod{N}(real(x)) + GaussMod{N}(imag(x))*im
-
-Mod{N}(a::S,b::T) where {N,S<:Integer,T<:Integer} = GaussMod{N}(a+b*im)
-
-modulus(x::GaussMod{T}) where T = T
-
-real(x::GaussMod{N}) where N = Mod{N}(real(x.val))
-real(x::Mod) = x 
-
-imag(x::GaussMod{N}) where N = Mod{N}(imag(x.val))
-imag(x::Mod{N}) where N = Mod{N}(0)
+const GaussMod{N,T} = Mod{N,Complex{T}}
+GaussMod{N}(x::T) where {N, T<:Integer} = Mod{N,Complex{T}}(x)
+GaussMod{N}(x::T) where {N, T<:Complex} = Mod{N,T}(x)
+Mod{N}(x::Complex{Rational{T}}) where {N,T} = Mod{N}(real(x)) + Mod{N}(imag(x))*im
```

Comment:
missing test case.

Reply:
Hi, wondering where did you see the test coverage? Can I change the travis CI to github actions and add the code coverage in this PR so that I can check the coverage more carefully?

TYI: Travis CI is becoming very restrictive on the amount of jobs one can submit, so people are switching to Github actions.

Reply:
I did not "see" the missing test coverage, but did not find a test case in the test files, and I did not find a link to test coverage results in the README.

If you want to switch to github - CI, I give you a link to my current package; see directores `.ci`, `.github`, `.codecov.yml` and `README.md`. Most of the files in there have to be customized - but that is easy.

Probably the github actions have to be activated by the repository owner.

Reply:
Yeah, let's do it in a seperate PR.

Reply:
BTW: If you run the tests locally, ther is a coverage option:
```
]activate Mods
test --coverage
```
which leaves `*.jl.cov` files in `src`.